ModularTegustation/tegu_mobs/Wildhunt.dm

```diff
@@ -0,0 +1,568 @@
+	health = 10000
+	maxHealth = 10000
+	damage_coeff = list(RED_DAMAGE = 0.4, WHITE_DAMAGE = 0.8, BLACK_DAMAGE = 0.2, PALE_DAMAGE = 0.8)
+	attack_verb_continuous = "slices"
+	attack_verb_simple = "slice"
+	attack_sound = 'ModularTegustation/Tegusounds/Erlking/Erlkinghit.ogg'
+	icon_state = "Erlking"
+	icon_living = "Erlking"
+	icon_dead = "Erlking"
+	icon = 'ModularTegustation/Teguicons/WildHuntBoss.dmi'
+	faction = list("hostile")
+	mob_biotypes = MOB_ORGANIC|MOB_HUMANOID
+	light_range = 3
+	movement_type = GROUND
+	is_flying_animal = FALSE
+	speak_emote = list("says")
+	melee_damage_type = BLACK_DAMAGE
+	melee_damage_lower = 75 // Assuming everyone has at least 0.5 red armor
+	melee_damage_upper = 85
+	stat_attack = HARD_CRIT
+	rapid_melee = 2
+	ranged = TRUE
+	ranged_cooldown_time = 5
+	speed = 2
+	move_to_delay = 3
+	pixel_x = -8
+	base_pixel_x = -1
+	blood_volume = BLOOD_VOLUME_NORMAL
+	gps_name = "NTAF-V"
+	del_on_death = FALSE
+	death_message = "falls to the ground, decaying into glowing particles."
+	death_sound = 'ModularTegustation/Tegusounds/claw/death.ogg'
+	footstep_type = FOOTSTEP_MOB_HEAVY
+	del_on_death = TRUE
+
+	var/special_attacking = FALSE // Are you currently performing a special attack
+	var/special_windup = 8 // How many deciseconds between showing a tell for a special attack and using it
+	var/dullahan_speed = 1.75
+	var/current_stage = 1 //changes behaviour slightly on phase 2
+	var/stage_threshold = 5000 // enters stage 2 at or below this threshold
+	var/slam_cooldown
+	var/slam_cooldown_time = 20 SECONDS
+	var/slam_damage = 250
+	var/can_act = TRUE
+	var/charging = FALSE
+	var/dash_num = 10//the length of the dash, in tiles
+	var/dash_cooldown = 0
+	var/dash_cooldown_time = 4 SECONDS
+	var/list/been_hit = list() // Don't get hit twice.
+	var/throw_timer = 0
+	var/throw_cooldown = 11 SECONDS
+	var/throw_amount = 10 // How many blades to throw at once
+	var/throw_cone = 25 // Total firing angle of all red's projectiles.
+	var/throw_damage = 80 // Damage of each thrown blade
+	var/nightmare_mode = FALSE
+	var/pulse_cooldown
+	var/pulse_cooldown_time = 200 SECONDS
+	var/spawn_time
+	var/spawn_cooldown
+	var/spawn_cooldown_time = 60 SECONDS
+	var/spawn_time_cooldown
+	var/list/spawned_mobs = list()
+
+/mob/living/simple_animal/hostile/megafauna/Erlking/Initialize(mapload)
+	. = ..()
+	var/list/units_to_add = list(
+		/mob/living/simple_animal/hostile/Catherine = 1
+		)
+	AddComponent(/datum/component/ai_leadership, units_to_add, 1, TRUE, TRUE)
+
+
+
+/mob/living/simple_animal/hostile/megafauna/Erlking/OpenFire()
+	if(!can_act)
+		return FALSE
+
+	if(dash_cooldown <= world.time)
+		var/chance_to_dash = 50
+		if(prob(chance_to_dash))
+			Erlking_dash(target)
+
+	if((slam_cooldown <= world.time))
+		CoffinSlam(target)
+		return
+
+	if(((throw_timer <= world.time)))
+		switch(current_stage)
+			if(2)
+				if(isliving(target))
+					return BladeThrow(target)
+		return
+	return

```

Comment:
what the actual fuck is going on here

Reply:
To be completely fair, this enemy just kills you haha